src/shared/options/readOptions.ts

```diff
@@ -35,42 +36,55 @@ export async function readOptions(args: string[]): Promise<OptionsParseResult> {
-		base: values.base as InputBase,
-		createRepository: values["create-repository"] as boolean | undefined,
-		description: values.description as string | undefined,
-		email: values.email as string | undefined,
-		excludeCompliance: values["exclude-compliance"] as boolean | undefined,
-		excludeContributors: values["exclude-contributors"] as boolean | undefined,
-		excludeLintJson: values["exclude-lint-json"] as boolean | undefined,
-		excludeLintKnip: values["exclude-lint-knip"] as boolean | undefined,
-		excludeLintMd: values["exclude-lint-md"] as boolean | undefined,
-		excludeLintPackageJson: values["exclude-lint-package-json"] as
-			| boolean
-			| undefined,
-		excludeLintPackages: values["exclude-lint-packages"] as boolean | undefined,
-		excludeLintPerfectionist: values["exclude-lint-perfectionist"] as
-			| boolean
-			| undefined,
-		excludeLintSpelling: values["exclude-lint-spelling"] as boolean | undefined,
-		excludeLintYml: values["exclude-lint-yml"] as boolean | undefined,
-		excludeReleases: values["exclude-releases"] as boolean | undefined,
-		excludeRenovate: values["exclude-renovate"] as boolean | undefined,
-		excludeTests: values["unit-tests"] as boolean | undefined,
-		funding: values.funding as string | undefined,
-		owner: values.owner as string | undefined,
-		repository: values.repository as string | undefined,
+	const mappedOptions = {
+		author: values.author,
+		base: values.base,
+		createRepository: values["create-repository"],
+		description: values.description,
+		email: values.email,
+		excludeCompliance: values["exclude-compliance"],
+		excludeContributors: values["exclude-contributors"],
+		excludeLintJson: values["exclude-lint-json"],
+		excludeLintKnip: values["exclude-lint-knip"],
+		excludeLintMd: values["exclude-lint-md"],
+		excludeLintPackageJson: values["exclude-lint-package-json"],
+		excludeLintPackages: values["exclude-lint-packages"],
+		excludeLintPerfectionist: values["exclude-lint-perfectionist"],
+		excludeLintSpelling: values["exclude-lint-spelling"],
+		excludeLintYml: values["exclude-lint-yml"],
+		excludeReleases: values["exclude-releases"],
+		excludeRenovate: values["exclude-renovate"],
+		excludeTests: values["unit-tests"],
+		funding: values.funding,
+		owner: values.owner,
+		repository: values.repository,
 		skipGitHubApi: !!values["skip-github-api"],
 		skipInstall: !!values["skip-install"],
 		skipRemoval: !!values["skip-removal"],
-		skipRestore: values["skip-restore"] as boolean | undefined,
+		skipRestore: values["skip-restore"],
 		skipUninstall: !!values["skip-uninstall"],
-		title: values.title as string | undefined,
+		title: values.title,
 	};
 
+	const optionsParseResult = optionsSchema.safeParse(mappedOptions);
+
+	if (!optionsParseResult.success) {
+		return {
+			cancelled: true,
+			options: Object.entries(mappedOptions).reduce(
+				(acc, [key, value]) => ({
+					...acc,
+					[key]: typeof value === "string" ? undefined : value,
+				}),
+				{},
+			),
```

Comment:
Yeah it's a good question. Both in dev / type system land and for runtime reporting.

For type system land: right now `OptionsParseCancelled` assumes `options: Partial<Option>`. But Zod is pointing out to us that that's not true - and never was. I think changing `OptionsParseCancelled.options` to `object` would be the right step in the type system, to convey that _"we don't know what `options` actually is, other than that it's some object"_. That way there's no need for this `.reduce`.

For runtime reporting: how about printing with [`zod-error`](https://www.npmjs.com/package/zod-error)?

Putting them together: maybe the `OptionsParseCancelled` type should somehow also include the Zod error if there is one? Some massaging of the types so that `bin/index.ts` knows how to print an invalid inputs error from Zod?